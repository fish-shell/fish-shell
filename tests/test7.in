# Test that subsequent cases do not blow away the status from previous ones
for val in one two three four
	switch $val
	case one
		/bin/sh -c 'exit 1'
	case two
		/bin/sh -c 'exit 2'
	case three
		/bin/sh -c 'exit 3'
	end
	echo $status
end

echo

# Test that the `switch` builtin itself does not blow away status before evaluating a case
false
switch one
case one
	echo $status
end

# Test that non-case tokens inside `switch` don't blow away status
# (why are these even allowed?)
false
switch one
true
case one
	echo $status
end

#test contains -i
echo test contains -i
contains -i string a b c string d
contains -i string a b c d; or echo nothing
contains -i -- string a b c string d
contains -i -- -- a b c; or echo nothing
contains -i -- -- a b c -- v
