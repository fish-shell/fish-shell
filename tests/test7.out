1
2
3
0

1
test contains -i
4
nothing
4
nothing
4
alpha1.1
alpha1.2
beta2.1
beta2.2
yep3.1
yep3.2
gamma3.1
gamma3.2
yep4.1
yep4.2
delta4.1
delta4.2
epsilon5.2
zeta 6.2
###
0
1
